### PR TITLE
Close #13810: Refactor TrackDesignManage window to class

### LINKED ISSUE #13810: Refactor window to class: TrackDesignManage
See #13737 for more details.

### PR DESCRIPTION
**Pull request by tupaschoal**

### CODE REVIEW

File: src/openrct2-ui/windows/TrackDesignManage.cpp [hunk 1]
```diff
@@ -125,41 +121,34 @@ static void WindowTrackManageClose(rct_window* w)
...
-static void WindowTrackManageTextinput(rct_window* w, WidgetIndex widgetIndex, char* text)
+void TrackDesignManageWindow::OnTextInput(WidgetIndex widgetIndex, std::string_view text)
 {
-    if (widgetIndex != WIDX_RENAME || str_is_null_or_empty(text))
+    if (widgetIndex != WIDX_RENAME || text.empty())
     {
         return;
     }
 
-    if (str_is_null_or_empty(text))
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignManage.cpp:162]: This part didn't make a lot of sense to me, hence why I added a TODO

Comment (Gymnasiast) [reply src/openrct2-ui/windows/TrackDesignManage.cpp:162]: It was added in https://github.com/OpenRCT2/OpenRCT2/commit/e666028e0274be0c3c6195e691c96ffcfb897cd5 , most likely to add a specific error message for this case. It looks like an oversight. I think you can simply get rid of the second case on line 157/145.

File: src/openrct2-ui/windows/TrackDesignManage.cpp [hunk 2]
```diff
@@ -183,51 +169,48 @@ static void WindowTrackManageTextinput(rct_window* w, WidgetIndex widgetIndex, c
...
-/**
- *
- *  rct2: 0x006D3523
- */
-static void WindowTrackManagePaint(rct_window* w, rct_drawpixelinfo* dpi)
+void TrackDesignManageWindow::OnDraw(rct_drawpixelinfo& dpi)
 {
     Formatter::Common().Add<const utf8*>(_trackDesignFileReference->name.c_str());
-    WindowDrawWidgets(*w, dpi);
+    WindowDrawWidgets(*this, &dpi);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TrackDesignManage.cpp:175]:
```suggestion
    DrawWidgets(dpi);
```

File: src/openrct2-ui/windows/TrackDesignManage.cpp [hunk 3]
```diff
@@ -240,18 +223,14 @@ static void WindowTrackDeletePromptMouseup(rct_window* w, WidgetIndex widgetInde
...
 
-/**
- *
- *  rct2: 0x006D37EE
- */
-static void WindowTrackDeletePromptPaint(rct_window* w, rct_drawpixelinfo* dpi)
+void TrackDeletePromptWindow::OnDraw(rct_drawpixelinfo& dpi)
 {
-    WindowDrawWidgets(*w, dpi);
+    WindowDrawWidgets(*this, &dpi);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TrackDesignManage.cpp:228]:
```suggestion
    DrawWidgets(dpi);
```

**Review by Broxzier — CHANGES_REQUESTED** [review]
Tested the build, and things seem to still work as expected. Changes look mostly good, only found two cases that need updating.

**Review by Broxzier — APPROVED** [review]

**Pull request merged**